File: app/src/javax/java/com/adawolfs/mvp/login/LoginContract.java
package com.adawolfs.mvp.login;

import com.adawolfs.mvp.mvp.BaseMvpInteractor;
import com.adawolfs.mvp.mvp.BaseMvpView;
import com.adawolfs.mvp.mvp.BaseMvpPresenter;

import javax.inject.Inject;

import static android.R.attr.name;

/**
 * Created by adawolfs on 1/20/17.
 */

public class LoginContract {


    interface View extends BaseMvpView {

        void showProgress();

        void hideProgress();

        void setUsernameError();

        void setPasswordError();

        void navigateToHome();

    }

    interface Presenter extends BaseMvpPresenter<View> {
        void validateCredentials(String user, String password);
    }

    interface Interactor extends BaseMvpInteractor{
        void login(String user, String password);
    }

}

File: app/src/kotlin/java/com/adawolfs/mvp/login/LoginContract.kt
package com.adawolfs.mvp.login

import com.adawolfs.mvp.mvp.BaseMvpInteractor
import com.adawolfs.mvp.mvp.BaseMvpPresenter
import com.adawolfs.mvp.mvp.BaseMvpView

/**
 * Created by adawolfs on 1/20/17.
 */
class LoginContract{

    interface View : BaseMvpView {

        fun showProgress()

        fun hideProgress()

        fun setUsernameError()

        fun setPasswordError()

        fun navigateToHome()

    }

    interface Presenter: BaseMvpPresenter<View> {
        fun validateCredentials(user : String, password : String)
    }

    interface Interactor : BaseMvpInteractor{
        fun login(user: String, password: String)
    }
}